src/Core/Generator/Sampler/CosmosExecutor.cs

```diff
@@ -53,7 +53,15 @@ public async Task<List<T>> ExecuteQueryAsync<T>(string query, Action<T?>? callba
                     string content = await sr.ReadToEndAsync();
 
                     using JsonDocument jsonDocument = JsonDocument.Parse(content);
-                    JsonElement root = jsonDocument.RootElement.GetProperty("Documents");
+
+                    JsonElement root = jsonDocument.RootElement;
+                    if (jsonDocument.RootElement.ValueKind != JsonValueKind.Array)
+                    {
```

Comment:
The PR description says that it allows schema generation if "an empty array was present in the JSON." however you are checking for value kind != Array here, doesnt explain how this fixes empty array situation.. can you please explain?

Reply:
Or is this code fixing something else? If yes, what is it trying to accomplish? Can you provide some valid examples of supported jsonDocuments here?

Reply:
this is the fix for that https://github.com/Azure/data-api-builder/blob/dc872a612c084cf1bcd30bd5612f6fb0a117e465/src/Core/Generator/SchemaGenerator.cs#L279, previously if array is empty, it was messing up the logic. After this fix, it won't process empty array at all.